lib/ddtrace/contrib/delayed_job/patcher.rb

```diff
@@ -0,0 +1,45 @@
+module Datadog
+  module Contrib
+    module DelayedJob
+      # DelayedJob integration
+      module Patcher
+        include Base
+        register_as :delayed_job
+
+        option :service_name, default: 'delayed_job'.freeze
+        option :tracer, default: Datadog.tracer
+
+        @patched = false
+
+        class << self
+          def patch
+            return @patched if patched? || !defined?(::Delayed)
+
+            require 'ddtrace/ext/app_types'
+            require_relative 'plugin'
+
+            add_instrumentation(::Delayed::Worker)
+            @patched = true
+          rescue => e
+            Tracer.log.error("Unable to apply DelayedJob integration: #{e}")
+            @patched
+          end
+
+          def patched?
+            @patched
+          end
+
+          private
+
+          def unpatch
```

Comment:
Consider that this patch is not actually "unpatchable". We should not communicate the expectation that you can "undo" patches, or that you should be able to "undo" patches, since modifying classes and constants is generally irreversible. Trying to support this would be useful for tests, but only for tests; if this kind of feature only needs to exist for tests, then in should exist in the test suite, not in our patcher API.

Reply:
good point